src/classes/PS_CommitmentRequest.cls

```diff
@@ -209,14 +209,36 @@ public inherited sharing class PS_CommitmentRequest {
     * @param token Payment Method Token
     * @return RequestBody
     */
-    private RequestBody getUpdateRequestBody(npe03__Recurring_Donation__c rd, String token) {
+    private RequestBody getUpdateRequestBody(npe03__Recurring_Donation__c rd, npe03__Recurring_Donation__c oldRd, String token) {
         RequestBody reqBody = new RequestBody()
             .withCommitmentId(rd.CommitmentId__c)
             .withPaymentMethodToken(token);
 
+        String currencyCode = UserInfo.isMultiCurrencyOrganization()
+            ? (String) rd.get('CurrencyIsoCode')
+            : UserInfo.getDefaultCurrency();
+
+        if (isElevateScheduleFieldsChanged(rd, oldRd)) {
+            reqBody.deactivateOldSchedules();
+            setSchedules(reqBody, scheduleService.buildNewSchedules(rd), currencyCode);
+        }
+        
         return reqBody;
     }
 
+    /**
+    * @description Check if Elevate schedule needs to be update
+    * @param rd new Recurring Donation
+    * @param oldRd old Recurring Donation
+    */
+    public Boolean isElevateScheduleFieldsChanged(npe03__Recurring_Donation__c rd, npe03__Recurring_Donation__c oldRd) {
```

Comment:
Can we rename this to `isAmountFieldChanged`?

Reply:
I prefer isScheduleFieldsChanged because Amount is part of the the Schedule on the Elevate side.  The Interval/Frequency/Period change will also be part of this method in the upcoming WI.

Reply:
Will it eventually be using some instance variables/methods too at that point? This method could be static, and I think it might be worth moving it out of `PS_CommitmentRequest` if it is made static.